corelog/formatters.go

```diff
@@ -0,0 +1,103 @@
+package corelog
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"runtime"
+	"strings"
+	"time"
+)
+
+var pid = os.Getpid()
+
+type Formatter interface {
+	Format(pkg string, level LogLevel, depth int, entries ...LogEntry)
+	SetWriter(w io.Writer)
+}
+
+type StringFormatter struct {
+	w *bufio.Writer
+}
+
+func (s *StringFormatter) Format(pkg string, _ LogLevel, _ int, entries ...LogEntry) {
+	s.w.WriteString(pkg)
```

Comment:
Personally I like logging file:linenum (long form) or nothing (short form). Printing pkg seems like it'll take up the space but not provide enough info to quickly pinpoint the origin of the msg.